File: 3158/main.go
package main

import "fmt"

func main() {
	fmt.Println(duplicateNumbersXOR([]int{1,2,1,3}))
	fmt.Println(duplicateNumbersXOR([]int{1,2,2, 1}))
}

func duplicateNumbersXOR(nums []int) int {
	ans := 0 
	m := make(map[int]int)
	for _, i := range nums {
		m[i]++
	}
	for k, v := range m {
		if v > 1 {
			ans ^= k
		}
	}
	return ans
}

File: 3158/main.rs
use std::collections::HashSet;

struct Solution {}

impl Solution {
    pub fn duplicate_numbers_xor(nums: Vec<i32>) -> i32 {
        let mut cnt = HashSet::new();
        let mut ans = 0;
        for i in nums {
            if cnt.contains(&i) {
                ans ^= i;
            } else {
                cnt.insert(i);
            }
        }
        ans
    }
}

fn main() {
    println!("{}", Solution::duplicate_numbers_xor(vec![1,2,2,1]));
}